planet/http.py

```diff
@@ -93,11 +93,13 @@ def _convert_and_raise(cls, error):
 
     @classmethod
     def _raise_for_status(cls, response):
-        try:
-            response.raise_for_status()
-        except httpx.HTTPStatusError as e:
-            e.response.read()
-            cls._convert_and_raise(e)
+        if response.status_code >= 400:
```

Comment:
Unless we do this we will trip on the fact that `httpx.Response(302).raise_for_status()` raises an exception in version 0.23, even when a redirect is followed. I wonder if httpx isn't updating the status code or request after following a redirect? Might be worth checking the project's bug tracker about this.

Reply:
Good news is that `httpx.Response(302).is_error` is `False`.